src/app/authentication/authentication.component.ts

```diff
@@ -1,15 +1,74 @@
 import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
 
 @Component({
   selector: 'app-authentication',
   templateUrl: './authentication.component.html',
   styleUrls: ['./authentication.component.scss']
 })
 export class AuthenticationComponent implements OnInit {
+  isLogin = true;
+  authForm: FormGroup;
+  label_button_up = 'Login';
+  label_button_down = 'Sign Up';
+  badEmail = false;
 
-  constructor() { }
+  constructor(private formBuilder: FormBuilder) {
+    this.authForm = this.formBuilder.group({
+      user: this.formBuilder.group({
+        first_name: '',
+        last_name: '',
+        email: [null, Validators.compose([Validators.required, Validators.email])],
+        password: [null, Validators.compose([Validators.required, Validators.pattern('[a-zA-Z]+[0-9]+')])],
+        password_confirmation: [''],
+        locale: 'en'
+      }, {
+        validator: this.mustMatch('password', 'password_confirmation')
+      })
+    });
+  }
 
   ngOnInit(): void {
   }
 
+  onSubmit(value) {
+    value.user.locale = 'en'
+    console.log(value);
+    this.authForm.reset();
+  }
+
+  checkInput(input: string) {
+    return !this.authForm.get('user')['controls'][input].valid && this.authForm.get('user')['controls'][input].touched;
+  }
+
+  checkPasswords(group: FormGroup) {
```

Comment:
```suggestion
  checkPasswords(group: : AbstractControl) {
```
Where do you use this function??